x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/process/MlMemoryTracker.java

```diff
@@ -110,11 +111,6 @@ private void setReassignmentRecheckInterval(TimeValue recheckInterval) {
     @Override
     public void onMaster() {
         isMaster = true;
-        try {
-            asyncRefresh();
-        } catch (Exception ex) {
-            logger.warn("unexpected failure while attempting asynchronous refresh on new master assignment", ex);
-        }
```

Comment:
I would prefer to leave this in (with the log message converted to `logIfNecessary`).  I'd prefer to do a second iteration of changes in a few months if we continue to see lots of errors rather than remove lots of refreshes in one go and find we've gone too far and caused other problems that are worse than log spam.

Reply:
> I'd prefer to do a second iteration of changes in a few months if we continue to see lots of errors rather than remove lots of refreshes in one go and find we've gone too far and caused other problems that are worse than log spam.

That sounds reasonable to me